### PR TITLE
DLEGION-cookoff-handledamage

### PR DESCRIPTION
**Pull request by DLEGION**
**When merged this pull request will:**
- Describe what this pull request will do
- Each change in a separate line
- Include documentation if applicable
- Respect the [Development Guidelines](http://ace3mod.com/wiki/development/)

PROPOSED EDIT BY DLEGION :
 * please be patient i'm a poor noob about code, i hardly understand it,
 * and there are sure better ways to get the same/better results,
 * i just do my best to get the desired results, and all the changes i will submit are tested,
 * and to my knowledge they work good with no problems / bugs.
 * thanks for your time spent on my suggestions.
 * COOKOFF SPEFIC EDIT INFO:
 * the problem with 3.7.0 version: Leopard AAF tank seems to be unkillable (not exploding) by Comanche DAGR and cannon,
 * by kajman AG missiles and cannon, and with titan AT rockets sometimes blow up with 1 hit, sometimes need 4 hits.
 * the solution : as temporary fix i found that changing some numbers and removing a couple of lines, 
 * now the Comanche DAGR and cannon, kajman AG missiles and 30mm cannon with AP rounds (and only AP rounds, HE don't harm it)
 * shot in the turret destroy the tank by cooking it off as they should. 
 * titan too seems to works correctly. only from some angles it dont kill it in 1 shot, but that's ok for me!

### CODE REVIEW

File: addons/cookoff/functions/fnc_handleDamage.sqf [hunk 1]
```diff
@@ -60,17 +73,13 @@ if (_simulationType == "tank") exitWith {
...
     private _ammoLocationHitpoint = getText (_vehicle  call CBA_fnc_getObjectConfig >> QGVAR(ammoLocation));
 
-    if (_hitIndex in (GVAR(cacheTankDuplicates) getVariable (typeOf _vehicle))) then {
-        _hitpoint = "#subturret";
-    };
-    
     // ammo was hit, high chance for cook-off
     if (_hitpoint == _ammoLocationHitpoint) then {
-        if (_damage > 0.5 && {random 1 < 0.7}) then {
+        if (_damage > 0.5 && {random 1 < 1.7}) then {
```
Comment (commy2) [inline addons/cookoff/functions/fnc_handleDamage.sqf]: that doesn't make sense. `random 1 < 1.7` is always `true`

File: addons/cookoff/functions/fnc_handleDamage.sqf [hunk 2]
```diff
@@ -60,17 +73,13 @@ if (_simulationType == "tank") exitWith {
...
             _vehicle call FUNC(cookOff);
         };
     } else {
-        if (_hitpoint in ["hithull", "hitturret", "#structural"] && {_newDamage > 0.6 + random 0.3}) then {
+        if (_hitpoint in ["hitbody", "hitturret", "#structural"] && {_newDamage > 0.5 + random 0.3}) then {
```
Comment (Glowbal) [inline addons/cookoff/functions/fnc_handleDamage.sqf]: This reintroduces the bug with incorrect hitpoints for tank

File: addons/cookoff/functions/fnc_handleDamage.sqf [hunk 3]
```diff
@@ -60,17 +73,13 @@ if (_simulationType == "tank") exitWith {
     // determine ammo storage location
     private _ammoLocationHitpoint = getText (_vehicle  call CBA_fnc_getObjectConfig >> QGVAR(ammoLocation));
 
-    if (_hitIndex in (GVAR(cacheTankDuplicates) getVariable (typeOf _vehicle))) then {
-        _hitpoint = "#subturret";
-    };
```
Comment (Glowbal) [inline addons/cookoff/functions/fnc_handleDamage.sqf]: why is this deleted?

File: addons/cookoff/functions/fnc_handleDamage.sqf (hunk 1 above)
Comment (Glowbal) [inline addons/cookoff/functions/fnc_handleDamage.sqf]: `random 1 < 1.7` means always true

**Comment by DLEGION** [comment]
im really sorry guys....i really dont know what i did exactly, i just know it works !
i really wish to help more, i'm sorry !

**Comment by DLEGION** [comment]
here is a video of a tank hit by an hellfire. its clear it MUST be destroyed ! :)

**Comment by DLEGION** [comment]
@Glowbal   heey man i fianlly had time to test the bug you mentioned about turrets, and you're right, few shots of 0.50 cal to the things on the turret (even leopard tank) make it cookoff and explode!
i'll search anotehr way to fix this!

**Comment by commy2** [comment]
those cannons you meantioned aren't supposed to be used against tanks.

**Comment by DLEGION** [comment]
well in fact only 30mm apfsds-t can succesfully penetrate up to 55mm or armor.....anyway lets consider this : a Comanche (that like DAGR were never adopted by an army in real life) its still an anti-tank helicopter, so it should have something able to defeat tanks. at least DAGR should be able to harm and destroy the MBT.
a solution can be equip with russian AG rockets renaming them hellfires, or solve this problem in other way, dunno.

**Comment by DLEGION** [comment]
....just a thought...probably its easier fix this by incrementing DAGR damage (indirectdamage range?)

**Comment by commy2** [comment]
You are changing stuff around just by your feelings at this point. There is no need to increase the damage of these rockets or exchange them for other ones.

**Comment by DLEGION** [comment]
exactly! i random change and try until i get a result ! ...and i get results! 
so i'm sure you guys that created this code will surely be able to fix this way better than me !
all i ask is to fig this game-breaking problems !
thanks !

**Comment by commy2** [comment]
I don't see the "game-breaking problems" though. Tanks are not supposed to blow up for no reason.

**Comment by DLEGION** [comment]
unlimited DAGR are not enought to blow up a tank ?  and what about specifically designed AG missiles like hellfires ?
seriously.....dont know whats wrong with you, but the issue is there. hope someone else will fix this.

**Comment by DLEGION** [comment]
i close this pull request as it needs more tweaking and testing.

**Comment by commy2** [comment]
@DLEGION the internals of the whole medical system are rewritten anyway atm

https://github.com/acemod/ACE3/projects/2

After we are done with this, we hope that it will be easier to maintain and tweak the whole system.
The rewrite also contains a change that should make damage more consistent overall and I hope that we can use that to adjust damage taken to work as good as possible.
Until then it's futile to work with the current code and configs in my opinion.
